File: tests/test_lc_346_mov_avg.py
import setup_paths

from lc_346_mov_avg import MovingAverage

def test_solve():

    #Test 1
    input   = [3,1,10,3,5]

    solution = MovingAverage(input[0])

    assert solution.next(input[1]) == 1.0
    assert solution.next(input[2]) == 5.5
    assert solution.next(input[3]) == 4.666666666666667
    assert solution.next(input[4]) == 6.0

File: tests/test_lc_346_mov_avg.c
#include <unity.h>
#include "lc_346_mov_avg.c"


void setUp()
{

}

void tearDown()
{

}

void test_MovingAverageCreate()
{
    //Test 1
    int input[] = {3,1,10,3,5};

    struct MovingAverage* my_ma = movingAverageCreate(input[0]);

    TEST_ASSERT (movingAverageNext(my_ma, input[1]) == 1.0);
    TEST_ASSERT (movingAverageNext(my_ma, input[2]) == 5.5);
    TEST_ASSERT (movingAverageNext(my_ma, input[3]) == 4.666666666666667);
    TEST_ASSERT (movingAverageNext(my_ma, input[4]) == 6.0);

}
